pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -17,151 +17,246 @@ limitations under the License.
+			expectPodToMirror: map[kubetypes.ResolvedPodUID]kubetypes.MirrorPodUID{
+				kubetypes.ResolvedPodUID("static-pod-uid"): kubetypes.MirrorPodUID("mirror-pod-uid"),
+			},
+			expectMirrorToPod: map[kubetypes.MirrorPodUID]kubetypes.ResolvedPodUID{
+				kubetypes.MirrorPodUID("mirror-pod-uid"): kubetypes.ResolvedPodUID("static-pod-uid"),
+			},
 		},
-	}
-
-	expectedPods := []*v1.Pod{
 		{
-			ObjectMeta: metav1.ObjectMeta{
-				UID:         "999999999",
-				Name:        "taco",
-				Namespace:   "default",
-				Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "api"},
+			name:                    "Get mirror pod",
+			podList:                 []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantUID:                 types.UID("static-pod-uid"),
+			wantPod:                 mirrorPod,
+			isMirrorPod:             true,
+			expectPods:              []*v1.Pod{staticPod, normalPod},
+			expectGetPod:            staticPod,
+			expectShouldBeMirrorPod: mirrorPod,
+			expectUID:               types.UID("static-pod-uid"),
+			expectPodToMirror: map[kubetypes.ResolvedPodUID]kubetypes.MirrorPodUID{
+				kubetypes.ResolvedPodUID("static-pod-uid"): kubetypes.MirrorPodUID("mirror-pod-uid"),
+			},
+			expectMirrorToPod: map[kubetypes.MirrorPodUID]kubetypes.ResolvedPodUID{
+				kubetypes.MirrorPodUID("mirror-pod-uid"): kubetypes.ResolvedPodUID("static-pod-uid"),
 			},
 		},
-		staticPod,
-	}
-	updates := append(expectedPods, mirrorPod)
-	podManager, _ := newTestManager()
-	podManager.SetPods(updates)
-
-	// Tests that all regular pods are recorded correctly.
-	actualPods := podManager.GetPods()
-	if len(actualPods) != len(expectedPods) {
-		t.Errorf("expected %d pods, got %d pods; expected pods %#v, got pods %#v", len(expectedPods), len(actualPods),
-			expectedPods, actualPods)
-	}
-	for _, expected := range expectedPods {
-		found := false
-		for _, actual := range actualPods {
-			if actual.UID == expected.UID {
-				if !reflect.DeepEqual(&expected, &actual) {
-					t.Errorf("pod was recorded incorrectly. expect: %#v, got: %#v", expected, actual)
-				}
-				found = true
-				break
-			}
-		}
-		if !found {
-			t.Errorf("pod %q was not found in %#v", expected.UID, actualPods)
-		}
-	}
-	// Tests UID translation works as expected. Convert static pod UID for comparison only.
-	if uid := podManager.TranslatePodUID(mirrorPod.UID); uid != kubetypes.ResolvedPodUID(staticPod.UID) {
-		t.Errorf("unable to translate UID %q to the static POD's UID %q; %#v",
-			mirrorPod.UID, staticPod.UID, podManager.mirrorPodByUID)
 	}
+	for _, test := range testCase {
+		t.Run(test.name, func(t *testing.T) {
+			podManager := newTestManager()
+			podManager.SetPods(test.podList)
+			actualPods := podManager.GetPods()
+			sort.Slice(actualPods, func(i, j int) bool { return actualPods[i].Name < actualPods[j].Name })
+			sort.Slice(test.expectPods, func(i, j int) bool { return test.expectPods[i].Name < test.expectPods[j].Name })
```

Comment:
```suggestion
```

Reply:
If we think the sorting code is too much, I think we should use cmp.diff, so that we only need to initialize sortOpt once like this https://github.com/kubernetes/kubernetes/pull/126895#discussion_r1730826388, and do not need too many sorting methods.

Reply:
It's not too much, but sorting set of pod refs for simple comparison is questionable as the tests work without it.